miriway.cpp

```diff
@@ -66,6 +68,20 @@ std::map<std::string, ShellCommands::CmdFunctor> const wm_command =
         { "exit", [](ShellCommands* sc, bool shift) { sc->exit(shift); } },
     };
 
+struct ShellComponentRunInfo
+{
+    time_t last_run_time = 0;
+    int runs_in_quick_succession = 0;
+    std::unique_ptr<miral::FdHandle> handle = nullptr;
+};
+
+std::string get_cmd_string(std::vector<std::string> const& cmd)
```

Comment:
This is really an implementation detail of `shell_launch` and seems a bit odd at namespace scope.

It could be moved into the lambda as a static lambda, or we could acknowledge that the lambda has "grown up" and create a corresponding explicit class with `get_cmd_string` and `ShellComponentRunInfo` as members along with the current `operator()`.

Reply:
I am cool with a static lambda for now